File: prop-all/src/test/java/io/pleo/prop/objects/MyAssistedInjectFactoryImp.java
package io.pleo.prop.objects;

import com.google.inject.Inject;
import com.google.inject.assistedinject.Assisted;
import io.pleo.prop.core.Prop;

import javax.inject.Named;

public class MyAssistedInjectFactoryImp {
  private final Prop<String> prop;
  private final String assistedArg;

  @Inject
  public MyAssistedInjectFactoryImp(
    @Named("io.pleo.test.prop3") Prop<String> prop,
    @Assisted String assistedArg
  ) {
    this.prop = prop;
    this.assistedArg = assistedArg;
  }
}

File: prop-all/src/test/kotlin/io/pleo/prop/objects/MyAssistedInjectFactoryImp.kt
package io.pleo.prop.objects

import com.google.inject.Inject
import com.google.inject.assistedinject.Assisted
import io.pleo.prop.core.Prop
import javax.inject.Named

@Suppress("unused")
class MyAssistedInjectFactoryImp
    @Inject
    constructor(
        @Named("io.pleo.test.prop3") private val prop: Prop<String>,
        @Assisted private val assistedArg: String,
    )
